src/compiler/parser.ts

```diff
@@ -4757,16 +4757,22 @@ module ts {
             var amdDependencies: string[] = [];
             var amdModuleName: string;
 
-            // Keep scanning all the leading trivia in the file until we get to something that 
-            // isn't trivia.  Any single line comment will be analyzed to see if it is a 
-            // reference comment.
+            // Have we found anything that isn't a comment or whitespace?
+            var hasEncounteredNonTrivia = false;
+
+            // Scan the file for single line comments to analyse for references comments
             while (true) {
                 var kind = triviaScanner.scan();
+
+                if (kind === SyntaxKind.EndOfFileToken) {
+                    break;
+                }
                 if (kind === SyntaxKind.WhitespaceTrivia || kind === SyntaxKind.NewLineTrivia || kind === SyntaxKind.MultiLineCommentTrivia) {
                     continue;
                 }
                 if (kind !== SyntaxKind.SingleLineCommentTrivia) {
-                    break;

```

Comment:
This effectively is scanning the whole file twice just to produce this error. The current behavior relies on scanning just a few tokens in the beginning of the file, which is manageable, more than that we are affecting build performance in a noticeable way. I am not sure it is worth it at this point. If we need to do this we have to do it as we are parsing, just to avoid the extra pass. 

@CyrusNajmabadi any ideas on how to report this error as we are parsing, in a way that also works with incremental parsing?